test/e2e/e2e_test.go

```diff
@@ -146,6 +146,19 @@ func TestE2EFlow(t *testing.T) {
 			VerifyResourcesProvisionedForSignup(t, awaitility, johnSignup, "basic")
 			VerifyResourcesProvisionedForSignup(t, awaitility, johnExtraSignup, "basic")
 		})
+
+		t.Run("delete useraccount and expect recreation", func(t *testing.T) {
+			// given
+			memberAwait := wait.NewMemberAwaitility(awaitility)
+	
+			// when deleting the user account
+			err := memberAwait.DeleteUserAccount(johnSignup.Status.CompliantUsername)
+	
+			// then the user account should be recreated
+			require.NoError(t, err)
+			_, err = memberAwait.WaitForUserAccount(johnSignup.Status.CompliantUsername)
```

Comment:
Wouldn't using `VerifyResourcesProvisionedForSignup()` be better?
`VerifyResourcesProvisionedForSignup` makes sure all the resources are in place and with proper status. Not just UserAcc.